dipy/reconst/ivim.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/python
+""" Classes and functions for fitting ivim model """
+from __future__ import division, print_function, absolute_import
+import warnings
+import functools
+import numpy as np
+from scipy.optimize import curve_fit
+
+from dipy.core.gradients import gradient_table
+from dipy.reconst.base import ReconstModel
+from dipy.reconst.dti import _min_positive_signal
+
+
+def ivim_function(bvals, S0, f, D_star, D):
+    S = S0 * (f * np.exp(-bvals * D_star) + (1 - f) * np.exp(-bvals * D))
+    return S
+
+
+class IvimModel(ReconstModel):
+    """Ivim model
+    """
+
+    def __init__(self, gtab, fit_method="NLLS", *args, **kwargs):
```

Comment:
Yes, I have followed the dti model and I thought that this was a good way to keep the fitting routines separate if we decide to have multiple fitting routines. If we decide to use only one routine [curve_fit or leastsq (which seems to be at the center of all fitting routines)] then I can insert the nlls_fit code here itself. I m now working on "leastsq" as a separate function similar to nlls fit. Once I complete the implementation and write the tests I can move the code in the fit method if we decide to use just one fitting routine.
What do you suggest ?